openconcept/weights/weights_jet_transport.py

```diff
@@ -0,0 +1,1827 @@
+            promotes_inputs=["ac|propulsion|engine|rating"],
+        )
+        self.add_subsystem(
+            "engine_multiplier",
+            ElementMultiplyDivideComp(
+                output_name="W_engines", input_names=["W_engine", "ac|propulsion|num_engines"], input_units=["lb", None]
+            ),
+            promotes_inputs=["ac|propulsion|num_engines"],
+            promotes_outputs=["W_engines"],
+        )
+        self.connect("single_engine.W_engine", "engine_multiplier.W_engine")
+
+        # -------------- Engine systems --------------
+        self.add_subsystem(
+            "engine_systems",
+            EngineSystemsWeight_JetTransport(),
+            promotes_inputs=[
+                "ac|propulsion|engine|rating",
+                "ac|propulsion|num_engines",
+                "ac|aero|Mach_max",
+                "ac|weights|W_fuel_max",
+            ],
+            promotes_outputs=[
+                "W_thrust_rev",
+                "W_eng_control",
+                "W_fuelsystem",
+                "W_eng_start",
+            ],
+        )
+        self.connect("single_engine.W_engine", "engine_systems.W_engine")
+
+        # -------------- Nacelle --------------
+        self.add_subsystem(
+            "nacelles",
+            NacelleWeight_JetTransport(),
+            promotes_inputs=["ac|propulsion|engine|rating", "ac|propulsion|num_engines"],
+            promotes_outputs=["W_nacelle"],
+        )
+
+        # ==============================================================================
+        # Furnishings for passengers
+        # ==============================================================================
+        self.add_subsystem(
+            "furnishings",
+            FurnishingWeight_JetTransport(K_lav=self.options["K_lav"], K_buf=self.options["K_buf"]),
+            promotes_inputs=[
+                "ac|num_passengers_max",
+                "ac|num_flight_deck_crew",
+                "ac|num_cabin_crew",
+                "ac|cabin_pressure",
+                "ac|weights|MTOW",
+            ],
+            promotes_outputs=["W_furnishings"],
+        )
+
+        # ==============================================================================
+        # Other equipment
+        # ==============================================================================
+        # Estimate the volume of the passenger cabin by treating it as a cylinder with
+        # the fuselage diameter and length of fuselage length times a constant factor
+        # (by default 0.75)
+        self.add_subsystem(
+            "cabin_volume",
+            om.ExecComp(
+                "V_pressurized = pi * 0.25 * fus_height * fus_height * fus_length * cabin_frac",
+                V_pressurized={"units": "ft**3", "val": 1},
+                fus_height={"units": "ft", "val": 1},
+                fus_length={"units": "ft", "val": 1},
+                cabin_frac={"val": self.options["cabin_length_frac"]},
```

Comment:
If `cabin_frac` is constant and not connected from upstream, you can add `constant`: True to remove it from AUTO_IVC (that's both good and bad so up to you)

Reply:
Oh cool, I didn't know about this feature